server/app.js

```diff
@@ -1,14 +1,33 @@
+const mongoose = require('mongoose');
 const express = require('express');
-const compression = require('compression');
-const controllers = require('./controllers/routes');
+const cors = require('cors');
+const bodyParser = require('body-parser');
+const logger = require('morgan');
+const parentRoute = require('./controllers/parent/routes.js');
+const eventRoute = require('./controllers/event/routes.js');
+const teacherRoute = require('./controllers/teacher/routes.js')
+require('env2')('../config.env');
 
+let DB_URL = process.env.DB_URL;
+const API_PORT = 3001;
 const app = express();
+app.use(cors());
 
-app.disable('x-powered-by');
-app.use(express.json());
-app.use(compression());
-app.use(controllers);
-app.set('port', process.env.PORT || 7777);
+mongoose.connect(DB_URL, { useNewUrlParser: true })
```

Comment:
can you have this connection thing in a separate file ? and call it only when you want .
what happening now is that you are opening connection to the database on each request no matter what the request is.